src/utils.ts

```diff
@@ -54,3 +54,15 @@ export function URI(url: string | URI = '') {
 
 export const isDictionary = (maybeDictionary: unknown): maybeDictionary is Dictionary<unknown> =>
   isObjectLike(maybeDictionary);
+
+export const getLocalRefValue = (document: unknown, $ref: string): unknown => {
+  return get(document, pointerToPath($ref));
+};
+
+export const maybeResolveLocalRef = (document: unknown, target: unknown): unknown => {
```

Comment:
1. I suggest to keep the name `resolveLocalRef`, since there's not a real external condition driving this and the early return feels more an optimisation rather than a real resolving condition
2. Since we're passing the target, we can be kind of confident that the returned object will still be of the target type — so I suggest this:
```suggestion
export const maybeResolveLocalRef = <T>(document: unknown, target: T): T => {
```

Reply:
Actually I don't think this is a good idea. The $ref can be pointing at anything - best to keep it as unknown, and do a specific guard check after attempting the resolve (which is what we do) to verify the final type.

Reply:
Hmm I am afraid I am not following:

```ts
type Test = {
    name: string,
    age: number,
    $ref?: string
}

declare function resolve<T>(target: T): Omit<T,'$ref'> & {[k:string]:unknown}

const t: Test = {
    name: 'Vincenzo',
    age: 31
};

const resolved = resolve(t);
```

Isn't the resolved *at least* a T?

Reply:
No - what if the $ref points to something that does not exist, or to some other random thing. It's why we do the specific guard check after the maybe resolve - https://github.com/stoplightio/http-spec/blob/fix/local-references/src/oas3/transformers/responses.ts#L18-L19.

Reply:
That's fair. Better be defensive — as long it's not `any` I'm ok with it.